src/components/RoadDamage/index.tsx

```diff
@@ -0,0 +1,261 @@
+
+export default function RoadDamageMap(): ReactElement {
+  const MapWithNoSSR = dynamic(() => import('./Map'), {
+    ssr: false
+  })
+
+  const { chainIds } = useUserPreferences()
+  const roadDamageAlgoChains: number[] = Object.keys(ROAD_DAMAGE_ALGO_DIDS).map(
+    (key) => Number(key)
+  )
+  const roadDamageAlgoDids: string[] = Object.values(ROAD_DAMAGE_ALGO_DIDS)
+
+  const { address: accountId } = useAccount()
+  const { data: signer } = useSigner()
+  const { autoWallet } = useAutomation()
+
+  const { fileName: resultFileName } = ROAD_DAMAGE_RESULT_ZIP
+  const [jobs, setJobs] = useState<ComputeJobMetaData[]>([])
+  const [refetchJobs, setRefetchJobs] = useState(false)
+  const [isLoadingJobs, setIsLoadingJobs] = useState(false)
+  const newCancelToken = useCancelToken()
+
+  const scrollToMapRef = useRef<any>()
+
+  const {
+    createOrUpdateRoadDamage,
+    roadDamageList,
+    deleteRoadDamage,
+    clearRoadDamages
+  } = useUseCases()
+
+  const [mapData, setMapData] = useState<RoadDamageUseCaseData[]>([])
+
+  useEffect(() => {
+    if (!roadDamageList) {
+      setMapData([])
+      return
+    }
+
+    setMapData(roadDamageList)
+  }, [roadDamageList])
+
+  const fetchJobs = useCallback(async () => {
+    if (!accountId) {
+      return
+    }
+
+    try {
+      setIsLoadingJobs(true)
+      // Fetch computeJobs for all selected networks (UserPreferences)
+      const computeJobs = await getComputeJobs(
+        chainIds,
+        accountId,
+        null,
+        newCancelToken()
+      )
+      if (autoWallet) {
+        const autoComputeJobs = await getComputeJobs(
+          chainIds,
+          autoWallet?.address,
+          null,
+          newCancelToken()
+        )
+        autoComputeJobs.computeJobs.forEach((job) => {
+          computeJobs.computeJobs.push(job)
+        })
+      }
+
+      setJobs(
+        // Filter computeJobs for dids configured in _constants
+        computeJobs.computeJobs.filter(
+          (job) =>
+            roadDamageAlgoDids.includes(job.algoDID) &&
+            job.status === 70 &&
+            job.results.filter((result) => result.filename === resultFileName)
+              .length > 0
+        )
+      )
+      setIsLoadingJobs(!computeJobs.isLoaded)
+    } catch (error) {
+      LoggerInstance.error(error.message)
+      setIsLoadingJobs(false)
+    }
+  }, [
+    chainIds,
+    roadDamageAlgoChains,
+    roadDamageAlgoDids,
+    accountId,
+    autoWallet,
+    newCancelToken
+  ])
+
+  useEffect(() => {
+    fetchJobs()
+  }, [refetchJobs, chainIds])
+
+  const addComputeResultToUseCaseDB = async (job: ComputeJobMetaData) => {
+    if (roadDamageList.find((row) => row.job.jobId === job.jobId)) {
+      toast.info('This compute job result already is part of the map view.')
+      return
+    }
+
+    const dataForSameInputExists =
+      roadDamageList.filter(
+        (row) =>
+          job.inputDID.filter((did) => row.job.inputDID.includes(did))
```

Comment:
question: Can we be sure that `inputDID` is never undefined?

Reply:
good catch, in reality this is always set after a compute job, but its not guaranteed by the interface. Added optional chaining.